sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -54,52 +51,28 @@ case class Mode(
     child: Expression,
     mutableAggBufferOffset: Int = 0,
     inputAggBufferOffset: Int = 0,
-    deterministicExpr: Expression = Literal.FalseLiteral)
+    isSortAsc: Option[Expression] = None)
```

Comment:
I think it's not worth let `isSortAsc` as the child expression of `Mode` due to it's none or foldable.